backend/db/manticore_db.py

```diff
@@ -354,89 +351,162 @@ async def create_custom_indices(self, for_space: str | None = None) -> None:
             return
-        
+
         for table in tables.get('data', []):
             await self.drop_index(table['Index'], True)
-                    
+
     async def list_indexes(self) -> set[str]:
-        tables = self.mc_command('SHOW TABLES') 
+        tables = self.mc_command('SHOW TABLES')
         if tables is None:
             return set()
-        
-        return set([item['Index'] for item in tables['data']])
 
+        return set([item['Index'] for item in tables['data']])
 
     async def drop_index(self, name: str, delete_docs: bool) -> bool:
         return bool(self.mc_command(f"DROP TABLE {name}"))
-        
-    
+
     async def search(
-        self, 
-        space_name: str, 
-        search: str, 
-        filters: dict[str, str | list | None], 
-        limit: int, 
-        offset: int, 
-        exact_subpath: bool = False, 
-        sort_type: SortType = SortType.ascending, 
-        branch_name: str | None = None, 
-        sort_by: str | None = None, 
-        highlight_fields: list[str] | None = None, 
-        schema_name: str = "meta", 
-        return_fields: list = []
+            self,
+            space_name: str,
+            search: str,
+            filters: dict[str, str | list | None],
+            limit: int,
+            offset: int,
+            exact_subpath: bool = False,
+            sort_type: SortType = SortType.ascending,
+            branch_name: str | None = None,
+            sort_by: str | None = None,
+            highlight_fields: list[str] | None = None,
+            schema_name: str = "meta",
+            return_fields: list = []
     ) -> tuple[int, list[dict[str, Any]]]:
-        return (0, [])
-    
-    async def aggregate(self, 
-        space_name: str, 
-        filters: dict[str, 
-        str | list | None], 
-        group_by: list[str], 
-        reducers: list[Any], 
-        search: str | None = None, 
-        max: int = 10, 
-        branch_name: str = settings.default_branch, 
-        exact_subpath: bool = False, 
-        sort_type: SortType = SortType.ascending, 
-        sort_by: str | None = None, 
-        schema_name: str = "meta", 
-        load: list = []
-    ) -> list[Any]:
-        return []
-    
-    async def get_count(self, 
-        space_name: str, 
-        schema_shortname: str, 
-        branch_name: str = settings.default_branch
-    ) -> int:
-        return 0
-    
-    async def free_search(self, 
-        index_name: str, 
-        search_str: str, 
-        limit: int = 20, 
-        offset: int = 0
-    ) -> list[dict[str, Any]]:
-        return []
-    
+        sql_str = "SELECT "
+        if len(highlight_fields) == 0:
+            sql_str += "*" if len(return_fields) == 0 else ",".join(return_fields)
+        else:
+            sql_str += "HIGHLIGHT({}, %s)" % ",".join(highlight_fields)
+
+        sql_str += f" FROM {space_name}__{branch_name}__{schema_name}"
+
+        if search:
+            sql_str += f" WHERE MATCH('{search}')"
```

Comment:
don't use the MATCH prefix, pass the search string as it's, let the client manage the query